src/plugins/twitter/client.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime, timedelta
+import requests
+from requests_oauthlib import OAuth1
+
+
+class TwitterClient:
+
+    """ Twitter client for searching tweets about DOIs.
+
+    Please note, this uses the Premium Twitter search API. For more details
+    visit https://developer.twitter.com/en/docs/tweets/search/api-reference
+    /premium-search.html
+    """
+
+    verify_url = 'https://api.twitter.com/1.1/account/verify_credentials.json'
+    search_url_base = 'https://api.twitter.com/1.1/tweets/search'
+
+    def __init__(
+            self,
+            app_key,
+            app_key_secret,
+            access_token,
+            access_token_secret,
+            label
+
+    ):
+        """ Instantiate the Twitter client.
+
+        Args:
+            app_key (str): key for your twitter app
```

Comment:
Extremely minor: I think parameter descriptions should end with a full stop.

Reply:
Yeah, I imagine this will happen a lot. Will update them as I see them - thanks :+1: